### PR TITLE
Add "company_subdomain" in intercom auth params

### PR DESCRIPTION
**Pull request by srikar0896**

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -27,10 +27,12 @@ def google_oauth2
...
   def intercom
     data = request.env["omniauth.auth"]
+    params = request.env["omniauth.params"]
+
     token = data.credentials.token
     app_id = data.extra.raw_info.app.id_code
 
-    company = CompanySetting.find_by_intercom_workspace_id!(app_id).company
+    company = Company.find_by_subdomain!(params["company_subdomain"])
     company.company_setting.intercom_access_token = token
```
Comment (pranav7) [inline app/controllers/users/omniauth_callbacks_controller.rb:38]: Let's also save their `app_id` in `company_setting.intercom_workspace_id`?

Comment (srikar0896) [reply app/controllers/users/omniauth_callbacks_controller.rb:38]: yes, but in the same branch? I thought the addition of that column, migration, and remove the duplicate code from `intercom_controller#sheets` to be done in a separate branch.

Comment (pranav7) [reply app/controllers/users/omniauth_callbacks_controller.rb:38]: It's already a column on that table! :)
https://github.com/pranav7/cadet-app/blob/e695bcf7b6ce1c64df5838380e9bb4ff0b87e62f/app/controllers/intercom_controller.rb#L58

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 2]
```diff
@@ -58,5 +63,6 @@ def stub_env_for_omniauth # rubocop:disable Metrics/MethodLength
         }
       }
     )
+    request.env["omniauth.params"] = { "company_subdomain" => company.subdomain }
```
Comment (pranav7) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb:66]:
```suggestion
    request.env["omniauth.params"] = { company_subdomain: company.subdomain }
```

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 3]
```diff
@@ -58,5 +63,6 @@ def stub_env_for_omniauth # rubocop:disable Metrics/MethodLength
         }
       }
     )
+    request.env["omniauth.params"] = { company_subdomain: company.subdomain }
```
Comment (pranav7) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]:
```suggestion
    request.env["omniauth.params"] = { "company_subdomain" => company.subdomain }
```

**Pull request merged**